web3.js/src/connection.ts

```diff
@@ -3666,6 +3670,11 @@ export class Connection {
     const preflightCommitment =
       (options && options.preflightCommitment) || this.commitment;
 
+    const maxRetries = options && options.maxRetries;
+
+    if (maxRetries) {
```

Comment:
Could `maxRetries` be zero? That would evaluate to false, which I think would be unexpected. Can you take a look and a test for this?

Reply:
Very true, silly mistake
- fixed below

Reply:
I think the option is still not applied when the value is zero (retries disabled)?